server/mux.go

```diff
@@ -67,127 +67,137 @@ func NewMux(opts MuxOpts, service Service) http.Handler {
 		hr.NotFound = http.StripPrefix(opts.Basepath, hr.NotFound)
 	}
 
+	EnsureViewer := func(next http.HandlerFunc) http.HandlerFunc {
+		return AuthorizedUser(service.UsersStore, opts.UseAuth, ViewerRoleName, opts.Logger, next)
+	}
+	EnsureEditor := func(next http.HandlerFunc) http.HandlerFunc {
+		return AuthorizedUser(service.UsersStore, opts.UseAuth, EditorRoleName, opts.Logger, next)
+	}
+	EnsureAdmin := func(next http.HandlerFunc) http.HandlerFunc {
+		return AuthorizedUser(service.UsersStore, opts.UseAuth, AdminRoleName, opts.Logger, next)
+	}
+
 	/* Documentation */
 	router.GET("/swagger.json", Spec())
 	router.GET("/docs", Redoc("/swagger.json"))
 
 	/* API */
 	// Sources
-	router.GET("/chronograf/v1/sources", service.Sources)
-	router.POST("/chronograf/v1/sources", service.NewSource)
+	router.GET("/chronograf/v1/sources", EnsureViewer(service.Sources))
+	router.POST("/chronograf/v1/sources", EnsureEditor(service.NewSource))
 
-	router.GET("/chronograf/v1/sources/:id", service.SourcesID)
-	router.PATCH("/chronograf/v1/sources/:id", service.UpdateSource)
-	router.DELETE("/chronograf/v1/sources/:id", service.RemoveSource)
+	router.GET("/chronograf/v1/sources/:id", EnsureViewer(service.SourcesID))
+	router.PATCH("/chronograf/v1/sources/:id", EnsureEditor(service.UpdateSource))
```

Comment:
@nhaugo can an Editor edit sources? or just an Admin and above? i believe they can but wanting to confirm

Reply:
confirmed: yes, editor.